### ISSUE TITLE
precompile failure on Julia 1.3 master

### ISSUE THREAD

**Issue by chriscoey** [lanl-ansi/PowerModels.jl#500]
Currently it seems I'm forced to use Julia 1.1 with PowerModels, because I can't get it to precompile on 1.3 master:
```
[ Info: Precompiling PowerModels [c36e90e8-916a-50a6-bd94-075b64ef4655]
ERROR: Error while loading expression starting at /home/coey/.julia/packages/JuMP/PbnIJ/src/JuMP.jl:229
caused by [exception 1]
syntax: malformed "import" statement
Stacktrace:
 [1] top-level scope at /home/coey/.julia/packages/JuMP/PbnIJ/src/JuMP.jl:229
 [2] eval at ./boot.jl:330 [inlined]
 [3] eval(::Expr) at /home/coey/.julia/packages/JuMP/PbnIJ/src/JuMP.jl:11
 [4] top-level scope at /home/coey/.julia/packages/JuMP/PbnIJ/src/JuMP.jl:230
 [5] include at ./boot.jl:328 [inlined]
 [6] include_relative(::Module, ::String) at ./loading.jl:1094
 [7] include(::Module, ::String) at ./Base.jl:31
 [8] top-level scope at none:2
 [9] eval at ./boot.jl:330 [inlined]
 [10] eval(::Expr) at ./client.jl:432
 [11] top-level scope at ./none:3
ERROR: Error while loading expression starting at /home/coey/.julia/packages/InfrastructureModels/KIMEa/src/InfrastructureModels.jl:5
caused by [exception 1]
Failed to precompile JuMP [4076af6c-e467-56ae-b986-b466b2749572] to /home/coey/.julia/compiled/v1.3/JuMP/DmXqY.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1253
 [3] _require(::Base.PkgId) at ./loading.jl:1013
 [4] require(::Base.PkgId) at ./loading.jl:911
 [5] require(::Module, ::Symbol) at ./loading.jl:906
 [6] include at ./boot.jl:328 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1094
 [8] include(::Module, ::String) at ./Base.jl:31
 [9] top-level scope at none:2
 [10] eval at ./boot.jl:330 [inlined]
 [11] eval(::Expr) at ./client.jl:432
 [12] top-level scope at ./none:3
ERROR: Error while loading expression starting at /home/coey/.julia/packages/PowerModels/58ZTV/src/PowerModels.jl:6
caused by [exception 1]
Failed to precompile InfrastructureModels [2030c09a-7f63-5d83-885d-db604e0e9cc0] to /home/coey/.julia/compiled/v1.3/InfrastructureModels/Xgkf2.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1253
 [3] _require(::Base.PkgId) at ./loading.jl:1013
 [4] require(::Base.PkgId) at ./loading.jl:911
 [5] require(::Module, ::Symbol) at ./loading.jl:906
 [6] include at ./boot.jl:328 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1094
 [8] include(::Module, ::String) at ./Base.jl:31
 [9] top-level scope at none:2
 [10] eval at ./boot.jl:330 [inlined]
 [11] eval(::Expr) at ./client.jl:432
 [12] top-level scope at ./none:3
ERROR: Failed to precompile PowerModels [c36e90e8-916a-50a6-bd94-075b64ef4655] to /home/coey/.julia/compiled/v1.3/PowerModels/CdeMI.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1253
 [3] _require(::Base.PkgId) at ./loading.jl:1013
 [4] require(::Base.PkgId) at ./loading.jl:911
 [5] require(::Module, ::Symbol) at ./loading.jl:906

```

**Comment by ccoffrin** [comment]
Julia v1.1 is the current release version, correct?  Generally we are only targeting support for the current and previous Julia releases.

Is this a feature request to also have support for the nightly build?

**Comment by pseudocubic** [comment]
This has to do with JuMP, PowerModels should work with nightly once JuMP v0.19 support (`moi-2` branch) is merged into master

**Comment by chriscoey** [comment]
ah great, I thought it was an odd problem. 

any ETA on JuMP 0.19 support?

**Comment by chriscoey** [comment]
> Julia v1.1 is the current release version, correct? Generally we are only targeting support for the current and previous Julia releases.
> 
> Is this a feature request to also have support for the nightly build?

no that would be too much to ask

**Comment by chriscoey** [comment]
in the meantime, is the moi-2 branch in a useable state currently? because I'll just use that if so

**Comment by ccoffrin** [comment]
The various moi branches should be ready for primetime across the various packages.  Upgrades are planned for next week.

**Comment by ccoffrin** [comment]
Closing this as Julia nightly is now passing on master.

For anyone reading this, see https://github.com/JuliaOpt/JuMP.jl/pull/1909 for updates on the JuMP v0.18 fix.
